File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        dict_f = {}

        for num in nums:
            if num in dict_f:
                dict_f[num]+=1
            else:
                dict_f[num]=1
        
        top_k = []

        top_k=[key for key, value in sorted(dict_f.items(),key = lambda item:item[1],reverse = True)[:k]]
        return top_k

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int,int>, vector<pair<int,int>> , greater<pair<int,int>>> pq;
        unordered_map<int,int> mpp;
        vector<int>ans;

        for(int i=0; i<nums.size(); i++){
            mpp[nums[i]]++;
        }

        for(auto i = mpp.begin(); i!=mpp.end(); i++){
            pq.push({i-> second , i-> first});
            if(pq.size() > k){
                pq.pop();
            }
        }
        while(!pq.empty()){
            ans.push_back(pq.top().second);
            pq.pop();
        }

        return ans;
    }
};